### DISCUSSION TITLE
Error: Missing lock for job 1. failed

### DISCUSSION THREAD

**Discussion by aqibamir** [taskforcesh/bullmq#1677] [Q&A]
While doing stress testing I get the following error when I'm running multiple workers. I'm not fetching jobs manually so I think there shouldn't be such error in the first place 

<img width="910" alt="Screenshot 2023-02-10 at 9 10 54 PM" src="https://user-images.githubusercontent.com/73932489/218140093-34f35fd4-3d6d-4999-b309-47ac28235e93.png">

I also get the following error along with this error. 
Error: connect ETIMEDOUT
<img width="879" alt="Screenshot 2023-02-10 at 9 13 26 PM" src="https://user-images.githubusercontent.com/73932489/218140757-b3829449-a59d-40c1-8d3e-3a571250d1e4.png">

**Comment by manast** [comment]
You are aware you did not ask anything right? 😬

**Reply by aqibamir** [reply]
Why am i getting  `Error: Missing lock for job 1. failed` ? @manast

**Reply by manast** [reply]
That is pretty difficult to answer considering you did not provide test code to understand the context.

**Reply by aqibamir** [reply]
### Worker Creation 

` // Creating a worker process to process jobs on Auth service.
const fire = async () => {

  try {

    // Creates a connection to Redis.
    const redisConnection = new IORedis({
      port: process.env.REDIS_PORT,
      host: process.env.REDIS_URL,
      maxRetriesPerRequest: null
    });


    // Reference for databse connection and auth Service class. The databse connection is set in this file and then
    // the same connection is used onwards.
    var mongo = await MongoClient.connect(MONGO_URL);

    // Alert
    console.log("Database is connected");

    // Creating a new worker instance
    const authWorker = new worker('auth', activities, redisConnection, mongo);

    // Start the worker instance and start listening on auth queue.
    await authWorker.run();

    console.log("Worker Started");
  }
  catch (err) {
    console.log(err);
    throw err;
  }
}

fire(); `

**Reply by aqibamir** [reply]
### Worker Class

`export class worker {

  constructor(queueName, activities, connection, mongo = null) {

    try {

      if (queueName != null && activities != null && connection != null) {

        // Stores the redis connection for this instance of worker process.
        this.connection = connection;

        // Assign the function that are going to be executed to the instance of this worker.
        this.activities = activities;

        this.queueName = queueName;

        this.mongo = mongo;

        // These are the options that are configured when adding to a queue.
        this.resultQueueOpts = {
          connection: this.connection,
        }

        // These are the options that are configured when adding a job to the queue.
        this.jobAddOptions = {
          removeOnComplete: {
            age: 60, // keep up to 1 minute
          },
          removeOnFail: {
            age: 24 * 3600, // keep up to 24 hours
          },
        }

      } else {

        throw "Incomplete worker argumnets";

      }
    } catch (err) {

      throw err;

    }
  }


  async run() {
    // Function starts the worker process. It starts to listen for jobs on the queue defined in "queueName"

    try {

      const WorkerOptions = {
        connection: this.connection
      }

      // Create new worker.
      this.worker = new Worker(
        this.queueName,
        this.handleJobs.bind(this),
        WorkerOptions
      );

      // Autorun is done by default in bullmq. So there is no need to explicitly use run function.
      if (!this.worker.isRunning()) await this.worker.run();

    } catch (err) {

      console.log(err);
      throw err;

    }
  }`

**Reply by aqibamir** [reply]
I'm adding 2000 jobs in the queue using a python script and running 3 workers.

**Reply by manast** [reply]
Sorry, I do not see anything wrong with that code. The best is if you can write a small test case that can be run and that reproduces the issue, without any third-party dependencies and post an issue.
